### PR TITLE
[New port] cjm-numerics-uint128 

### PR DESCRIPTION
**Pull request by cpsusie**
**Describe the pull request**

- #### What does your PR fix?  
  Adds the C++ 20 cjm-numerics-uint128 header-only library to the project's ports.

- #### Which triplets are supported/not supported? 
  NOTE: This is a C++20 ONLY library.  Support in any environment is contingent on sufficient C++20 support.   
  Generally, as long as the compiler used complies with the libraries requirements (see [Compiler and Architecture Requirements](https://github.com/cpsusie/cjm-numerics#compiler--architecture-requirements), it SHOULD work for any triplet.  Also, please note that this library is truly header only.  Nothing is built until a user references it in a project.   That said, the project's repository contains a unit testing application (not included in library) and an example code application.  The unit test application has been tested to compile and execute successfully with the following triplets (and C++20 enabled):
* x64-windows-static (MSVC)
* x86-windows-static (MSVC)
* x64-linux (G++ 10.0+, Clang 11.0+ on kubuntu 20.10, compiling with C++20 enabled and using libstdc++)

The example code application (again, not included), in addition to MSVC, is also compiled for Windows using the Clang (currently 11) and the most up-to-date LLVM-based Intel compiler with the following triplets:
* x64-windows-static
* x86-windows-static

- #### Have you updated the [CI baseline]
No.  I am unsure of how this would work given that installation of the library (as strictly header-only) does not involve the compilation of any code -- merely a copy of the headers and inline files to the appropriate include directory.  Also, I am unfamiliar with automated CI and would appreciate any appropriate direction.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Your answer`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  <Yes / I am still working on this PR>

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/cjm-numerics-uint128/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "cjm-numerics-uint128",
+  "version-string": "1.0.0.0",
```
Comment (JonLiu1993) [inline ports/cjm-numerics-uint128/vcpkg.json:3]:
```suggestion
  "version": "1.0.0.0",
```

File: ports/cjm-numerics-uint128/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  "description": [
+    "a header-only library for C++ 20 that provides an unsigned 128-bit integer.",
+    "This library provides constexpr support for most operations but uses std::is_constant_evaluated to ensure that a more efficient operation, if available, is used during runtime execution.  This uint128 type can serve as the basic building block for future fixed-but-extended-precision integral and rational types."
+  ],
```
Comment (JonLiu1993) [inline ports/cjm-numerics-uint128/vcpkg.json:7]:
```suggestion
  "description": a header-only library for C++ 20 that provides an unsigned 128-bit integer.
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
@cpsusie, after the modification is completed, please run the command "./vcpkg x-add-version cjm-numerics-uint128 --overwrite-version", and then submit the changes again.

**Comment by JonLiu1993** [comment]
Ping @cpsusie  for response, Is work still being done for this PR?

**Comment by cpsusie** [comment]
I intend to resolve the problems but a critical bug popped up on another project so my open source attention has been there.  When that other bug is resolved, I'll get back to getting uint128 in shape for vcpkg.  Not sure if should be via this PR or if better to open another one.



Sent from my Verizon, Samsung Galaxy smartphone



-------- Original message --------
From: JonLiu1993 ***@***.***>
Date: 8/3/21 5:10 AM (GMT-05:00)
To: microsoft/vcpkg ***@***.***>
Cc: Christopher Susie ***@***.***>, Mention ***@***.***>
Subject: Re: [microsoft/vcpkg] [New port] cjm-numerics-uint128 (#18793)


Ping @cpsusie<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fcpsusie&data=04%7C01%7C%7C24225460869d4a1e190108d9565e84ee%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637635786225502731%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=i1j0LdVJYgg4LJoBdwJN%2B6ic1qOruq76A1NT5DzcKcw%3D&reserved=0> for response, Is work still being done for this PR?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fvcpkg%2Fpull%2F18793%23issuecomment-891675829&data=04%7C01%7C%7C24225460869d4a1e190108d9565e84ee%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637635786225512729%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=cGR%2F8rLv0LNpSPB8E9HtnLmXaWUyVVhx2RxVCf34P58%3D&reserved=0>, or unsubscribe<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAIBZDN55NIKN77WE6RZCC3TT26W7ZANCNFSM47YQIFBA&data=04%7C01%7C%7C24225460869d4a1e190108d9565e84ee%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637635786225522723%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=j7JZhRqFdVZmTEptK9DP1uasOkAmMaVlglGbkHcvNg0%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7C%7C24225460869d4a1e190108d9565e84ee%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637635786225522723%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=RBZLTigQ9LpwEfgM9ukAGaw4oU9MbrCYfvKnwBNb18w%3D&reserved=0> or Android<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26utm_campaign%3Dnotification-email&data=04%7C01%7C%7C24225460869d4a1e190108d9565e84ee%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637635786225532711%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=oJXE%2Bk9dK3sLEmA6X7YejQvIicNu7ziqC0lLrcfwE%2Bw%3D&reserved=0>.

**Comment by JonLiu1993** [comment]
Pinging @cpsusie  for response. Is work still being done for this PR?

**Comment by JonLiu1993** [comment]
@cpsusie ,Is this PR being worked on?

**Comment by JonLiu1993** [comment]
I will close this pr, if you need to fix it, please reopen it

**Pull request closed without merging**